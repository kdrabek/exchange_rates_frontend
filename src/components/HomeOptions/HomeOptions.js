import React, { Component, PropTypes } from 'react';
import { Link } from 'react-router';
import { FormGroup, ControlLabel, Button } from 'react-bootstrap';
import  DatePicker  from 'react-bootstrap-date-picker';

import { plDayLabels, plMonthLabels } from '../../utils/plLocale';

class HomeOptions extends Component {

  notificationsButton() {
    return (
      <Link to="/notifications">
        <Button bsStyle="primary" block>Przejdź do notyfikacji</Button>
      </Link>
    );
  }

  render() {
    return (
      <div>
        <h3>Opcje</h3>
        <hr />
        <FormGroup>
          <ControlLabel>Wybierz datę:</ControlLabel>
          <DatePicker
            id="example-datepicker"
            dateFormat="YYYY/MM/DD"
            weekStartsOnMonday
            dayLabels={plDayLabels}
            monthLabels={plMonthLabels}
            onChange={this.props.handleChange}
            value={this.props.tableDate}
          />
        </FormGroup>
        {this.props.isAuthenticated ? this.notificationsButton() : ''}
      </div>
    );
  }
}

HomeOptions.propTypes = {

};

export default HomeOptions;