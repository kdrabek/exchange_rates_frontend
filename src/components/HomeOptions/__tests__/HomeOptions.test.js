import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import HomeOptions from '../HomeOptions';

describe('HomeOptions component', () => {

  it('matches the snapshot when user authenticated', () => {
    const component = renderer.create(
      <HomeOptions 
        isAuthenticated={true}
        tableDate={'2017-04-07'}
        handleChange={() => {}}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('matches the snapshot when user not authenticated', () => {
    const component = renderer.create(
      <HomeOptions 
        isAuthenticated={false}
        tableDate={'2017-04-07'}
        handleChange={() => {}}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});