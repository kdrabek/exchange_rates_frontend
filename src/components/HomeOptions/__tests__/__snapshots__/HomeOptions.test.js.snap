exports[`HomeOptions component matches the snapshot when user authenticated 1`] = `
<div>
  <h3>
    Opcje
  </h3>
  <hr />
  <div
    className="form-group">
    <label
      className="control-label"
      htmlFor={undefined}>
      Wybierz datę:
    </label>
    <span
      className="input-group"
      id="example-datepicker_group">
      <div
        style={
          Object {
            "position": "relative",
          }
        } />
      <input
        data-formattedvalue="2017/04/07"
        id="example-datepicker"
        name={undefined}
        type="hidden"
        value="2017-04-07T12:00:00.000Z" />
      <input
        autoFocus={false}
        className="form-control"
        disabled={false}
        id={undefined}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="YYYY/MM/DD"
        style={
          Object {
            "width": "100%",
          }
        }
        type="text"
        value="2017/04/07" />
      <span
        className="input-group-addon"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }>
        <div
          style={
            Object {
              "opacity": 1,
            }
          }>
          ×
        </div>
      </span>
    </span>
  </div>
  <a
    onClick={[Function]}
    style={Object {}}>
    <button
      className="btn btn-primary btn-block"
      disabled={false}
      type="button">
      Przejdź do notyfikacji
    </button>
  </a>
</div>
`;

exports[`HomeOptions component matches the snapshot when user not authenticated 1`] = `
<div>
  <h3>
    Opcje
  </h3>
  <hr />
  <div
    className="form-group">
    <label
      className="control-label"
      htmlFor={undefined}>
      Wybierz datę:
    </label>
    <span
      className="input-group"
      id="example-datepicker_group">
      <div
        style={
          Object {
            "position": "relative",
          }
        } />
      <input
        data-formattedvalue="2017/04/07"
        id="example-datepicker"
        name={undefined}
        type="hidden"
        value="2017-04-07T12:00:00.000Z" />
      <input
        autoFocus={false}
        className="form-control"
        disabled={false}
        id={undefined}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="YYYY/MM/DD"
        style={
          Object {
            "width": "100%",
          }
        }
        type="text"
        value="2017/04/07" />
      <span
        className="input-group-addon"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }>
        <div
          style={
            Object {
              "opacity": 1,
            }
          }>
          ×
        </div>
      </span>
    </span>
  </div>
  
</div>
`;
